crates/ruff_linter/src/rules/flake8_quotes/rules/check_string_quotes.rs

```diff
@@ -354,8 +354,62 @@ fn strings(
     diagnostics
 }
 
+/// A builder for the f-string range.
+///
+/// For now, this is limited to the outermost f-string and doesn't support
+/// nested f-strings.
+#[derive(Debug, Default)]
+struct FStringRangeBuilder {
+    start_location: TextSize,
+    end_location: TextSize,
+    nesting: u32,
+}
+
+impl FStringRangeBuilder {
+    fn visit_token(&mut self, token: &Tok, range: TextRange) {
+        match token {
+            Tok::FStringStart => {
+                if self.nesting == 0 {
+                    self.start_location = range.start();
+                }
+                self.nesting += 1;
+            }
+            Tok::FStringEnd => {
+                // We can assume that this will never overflow because we know
+                // that the program once parsed to a valid AST which means that
+                // the start and end tokens for f-strings are balanced.
+                self.nesting -= 1;
```

Comment:
This will overflow if `self.nesting` is already 0 (which could happen because we use `flatten()`). So I think we should use `saturating_sub` or test if `self.nesting == 1` -> set the `end_location` and decrement. Otherwise decrement only

Reply:
Yes, sorry. A bad copy paste mistake.